synthesis_alu/alu.cpp

```diff
@@ -0,0 +1,82 @@
+#include <iostream>
+#include <systemc.h>
+
+using namespace std;
+
+SC_MODULE(alu) {
+    sc_in<bool> clk;
+    sc_in<bool> reset;
+    sc_in<sc_bv<32>> rs1;
+    sc_in<sc_bv<32>> rs2;
+    sc_in<sc_bv<12>> imm;
+    sc_in<sc_bv<17>> instruction;
+
+    sc_out<sc_bv<32>> rd;
+
+    void do_alu() {
+        while (true) {
+            wait();
+            if (reset) {
+                rd.write(0);
+            } else {
+
+                sc_bv<7> opcode = instruction.read().range(6, 0);
+                sc_bv<10> funct73 = instruction.read().range(16, 7);
+                if (opcode == 0b0110011) {         // R-Type
+                    if (funct73 == 0b0000000000) { // ADD
+                        rd.write(rs1.read().to_int() + rs2.read().to_int());
```

Comment:
Sind diese to_int() wirklich notwendig? M.E. kann man in System mit Signalen und ports normal rechnen :-)

Reply:
Ich bekomm den Error, dass + als Operand nicht für sc_bv definiert ist